src/syrupy/extensions/amber/serializer.py

```diff
@@ -95,6 +99,23 @@ def read_file(cls, filepath: str) -> "SnapshotFossil":
 
     @classmethod
     def serialize(
+        cls,
+        data: "SerializableData",
+        *,
+        exclude: Optional["PropertyFilter"] = None,
+        matcher: Optional["PropertyMatcher"] = None,
+    ) -> str:
+        """
+        After serializing, new line control characters are normalised. This is needed
+        for interoperablity of snapshot matching between systems that do not use the
+        same new line control characters. Example snapshots generated on windows os
+        should not break when running the tests on a unix based system and vice versa.
+        """
+        serialized = cls._serialize(data, exclude=exclude, matcher=matcher)
+        return serialized.replace("\r\n", "\n").replace("\r", "\n")
```

Comment:
I think I'm having trouble reasoning around how this impacts intentional carriage returns in a snapshot. Are we saying users are just not able to snapshot carriage returns at all? No way to separate an intentional carriage return from one coming from the OS?

Reply:
All carriage returns are translated into python newlines `"\n"`. Basically if you have `"\r\r\n"` it should end up being `"\n\n"` across all platforms.